### PR TITLE
Add postprocessing

### LINKED ISSUE #419: Add option to reverse animation
**Is your feature request related to a problem? Please explain.**
I don't know if I can't find it, but it looks like there is no easy way to reverse an animation

**Describe the solution you'd like**
Maybe one could add an argument to the render function so it is changes from 
```
function render(
    video::Video;
    framerate = 30,
    pathname = "javis_$(randstring(7)).gif",
    liveview = false,
    streamconfig::Union{StreamConfig,Nothing} = nothing,
    tempdirectory = "",
    ffmpeg_loglevel = "panic",
    rescale_factor = 1.0,
)
    layers = video.layers
    objects = video.objects
    frames = preprocess_frames!(video)
```
to
```julia
function render(
    video::Video;
    framerate = 30,
    pathname = "javis_$(randstring(7)).gif",
    liveview = false,
    streamconfig::Union{StreamConfig,Nothing} = nothing,
    tempdirectory = "",
    ffmpeg_loglevel = "panic",
    rescale_factor = 1.0,
    reverse_animation = false
)
    layers = video.layers
    objects = video.objects
    frames = preprocess_frames!(video)
    if reverse_animation 
        frames = reverse(frames)
    end
```

**Describe alternatives you've considered**
One could also create a `reverse_render` function, I have no clue which is better.

I can make the PR if it makes sense.

### PR DESCRIPTION
**Pull request by gpucce**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [ ] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [ ] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [ ] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [ ] Did I properly add test dependencies to the `test` directory (if applicable)?
- [ ] Did I check relevant tutorials that may be affected by changes in this PR?
- [ ] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #419 

**How did you address these issues with this PR? What methods did you use?**
I add a postprocess argument to render which is supposed to be a function that gets called on a vector of frames as returned by `get_javis_frame`

### CODE REVIEW

**Comment by gpucce** [comment]
@Wikunia I will move on to add tests. While I do that can you check if it makes sense the way I made it? Thanks!

File: src/Javis.jl [hunk 1]
```diff
@@ -275,15 +276,35 @@ function render(
         tempdirectory = mktempdir()
     end
 
+    unprocessed_frames = Vector{Matrix{RGB{Normed{UInt8,8}}}}()
+
     filecounter = 1
-    @showprogress 1 "Rendering frames..." for frame in frames
+    @showprogress 1 "Drawing frames..." for frame in frames
         frame_image = convert.(RGB, get_javis_frame(video, objects, frame; layers = layers))
+        push!(unprocessed_frames, frame_image)
```
Comment (Wikunia) [inline src/Javis.jl:284]: I think saving them all into a vector might slow down the render function dramatically.
I'm wondering whether it makes more sense to call `postprocess` with the name of the tempdirectory folder instead. 
That would only work for `gif` though. Another option would maybe be to check if the `postprocess` function is the identity function. If that is the case then do nothing otherwise use the workflow you provided. However for simple postprocessing where one does a change for each frame and don't need all of them this current way would still potentially slow it down/fill up the memory.

**Comment by Wikunia** [comment]
Also make sure to merge the latest master into this as there were some changes to the `render` function.

File: src/Javis.jl (hunk 1 above)
Comment (gpucce) [reply src/Javis.jl:281]: Ok I thought the vector would be a catch but don't really know how to work around that. Let me try a bit more if I don't come up with anything I add it only as a frame by frame function and then for the reversing I add it as a special case with its own keyword argument maybe?

Comment (Wikunia) [reply src/Javis.jl:281]: Yeah the problem is that all the images need to be in memory for this. Which is probably fine for small short animations but for people who want to create YT videos with it it's probably not going to end up well :smile: 
Special case for reverse seems reasonable to me.
